lib/files.ncl

```diff
@@ -72,7 +92,12 @@ let regenerate_files | Files -> nix.derivation.Derivation
           }
             file_descr.materialisation_method
         in
-        nix-s%"regenerate_function "%{copy_command}" "%{file_descr.file}" "%{file_descr.target}""%
+        let shell_escape = fun path =>
+          path
+          |> std.string.replace "\\" "\\\\"
+          |> std.string.replace m%"'"% m%"\'"%
+        in
+        nix-s%"regenerate_function '%{copy_command}' '%{file_descr.file}' $'%{shell_escape file_descr.target}'"%
```

Comment:
I _think_ we should use `'` instead of `$'`. Since the string is coming from Nickel, Nickel is already going to expand escape sequences: if you write `let file = 'foo\nbar'`, the `\n` will be replaced by a new line, being equivalent to

```console
$ echo 'foo
quote>bar'
foo
bar
```
 
Using `$'..'` will also expand escape sequences somehow at a second level, that is `foo\\nbar` will be expanded to `foo\nbar` by Nickel, and here using `$'..'` will make a difference. But I feel like this isn't very useful since you would use `foo\nbar` to begin with. It would also mean that to correctly represent the filename `foo\bar` (where nothing is interpreted), you now need to write `foo\\\\bar` (interpreted as `foo\\bar` by "the first level", that is Nickel, and then as `foo\bar` by the shell string `$'..'`).

```suggestion
        nix-s%"regenerate_function '%{copy_command}' '%{file_descr.file}' '%{shell_escape file_descr.target}'"%
```

Reply:
Your argument is convincing. I have updated my PR to use `'…'` (single quotes) instead of `$'…'` (dollar-sign and single quotes) to quote commands.

Reply:
Ah, I take it back.

Upon testing, e.g.,, using `files."'".content = "…"` in an organist `project.ncl`, it appears that we need to use `$'…'` (dollar-sign and single quotes) instead of `'…'` (single quotes). For otherwise in  `'…'` (single quotes), we may need to expand every `'` (single quote) as `'"'"'` to close the single quote, use double-quotes to quote another single quote, then reopen the single quote.

Also, the generated script `regenerate-files` needs to make shellcheck happy, and for filenames with dollar-sign, e.g., `files."$pwd".content = "…"`, using `$'…'` (dollar-sign and single quotes) makes shellcheck happy but not `'…'` (single quotes), unless we suppress [SC2016](https://www.shellcheck.net/wiki/SC2016).

Reply:
Yes, you are right that shellcheck handles single-quotes (`'…'`) and dollar-sign and single-quotes (`$'…'`) differently.

Given this difference, if we want to use single-quotes (`'…'`), we need to suppress [SC2016](https://www.shellcheck.net/wiki/SC2016)—for example, to output an extra line `# shellcheck disable=SC2016` above every `regenerate_function` in `regenerate_one`. (Otherwise, shellcheck blocks the build of `regenerate-files` when there are dollar-signs `$` in filenames, such as `files."$one".content = "…"` in `project.ncl`.) 

I am fine with using either single-quotes (`'…'`) or dollar-sign and single-quotes (`$'…'`). Due to SC2016, I currently slightly prefer dollar-sign and single-quotes (`$'…'`).

Reply:
Fair enough. I still feel like using `$` is a indirect work-around schell check bad behavior but it's relatively easy to change afterward anyway.